src/python/pants/backend/python/target_types_rules.py

```diff
@@ -182,6 +185,57 @@ async def generate_targets_from_python_test_utils(
     )
 
 
+class GenerateTargetsFromPexBinariesFromEntryPoints(GenerateTargetsRequest):
+    generate_from = PexBinariesFromEntryPointsGeneratorTarget
+
+
+@rule
+async def generate_targets_from_pex_binaries_from_entry_points(
+    request: GenerateTargetsFromPexBinariesFromEntryPoints,
+    union_membership: UnionMembership,
+) -> GeneratedTargets:
+    generator_addr = request.generator.address
+    entry_points_field = request.generator[PexEntryPointsField].value or []
+    overrides = request.generator[OverridesField].flatten()
+
+    # @TODO: Check for overlap:
+    #   E.g. `path.to.app` specd as well as `app`, that should error
+    #   do this by resolving entry points before creating binaries so all are of the form
+    #   `path.to.app`. (But note that the override field might have `app`)
+
+    def create_pex_binary(entry_point_spec: str) -> PexBinary:
+        target_fields = {
+            field.alias: field.value
+            for field in request.generator.field_values.values()
+            if not isinstance(field, (PexEntryPointsField, OverridesField))
+        }
+        target_fields.update(overrides.pop(entry_point_spec, {}))
+
+        return PexBinary(
+            {PexEntryPointField.alias: entry_point_spec, **target_fields},
+            # TOREVIEW: We Should discuss the address, because users wil likely want to package or
+            #   run these scripts, so the address they use shouldn't be too weird.
+            # E.g. path/to/scriptdir:generator_name#foo.par:baz is kinda lengthy
```

Comment:
Reviewers, thoughts? It might be the best we can do.

Reply:
Yeah it's the best we can do. Especially because this design allows for multiple entry points on the same source file, e.g. `foo.py:main` and `foo.py:alt_main`.

Reply:
Yeah it's maybe a weird case, but we shouldn't assume it's never going to be in the wild :man_shrugging: